test/Microsoft.Health.Fhir.R4.Ingest.UnitTests/Template/CodeValueFhirTemplateProcessorTests.cs

```diff
@@ -542,5 +542,60 @@ public void GivenTemplateWithComponentAndObservationWithOutComponent_WhenMergObs
                         && v.end == boundary.end
                         && v.values.First().Item2 == "v2"));
         }
+
+        [Fact]
+        public void GivenTemplateWithCategory_ThenCategoryReturned()

```

Comment:
Maybe add another test where we merge observations with categories? Would help us solidify what we expect from that and ensure it's working as expected.